tembo-operator/tests/integration_tests.rs

```diff
@@ -2037,12 +2080,119 @@ mod test {
             }
             i += 1;
         }
-        // Get the ingress route tcp
         let ing_route_tcp = ingress_route_tcp_api.get(&ing_route_tcp_name).await;
-        // Should be deleted
         assert!(ing_route_tcp.is_err());
 
-        // Cleanup CoreDB resource
+        // Enable Dedicated Networking Test
+        let context = state.create_context(client.clone());
+
+        let coredb_json = serde_json::json!({
+            "apiVersion": API_VERSION,
+            "kind": kind,
+            "metadata": {
+                "name": name
+            },
+            "spec": {
+                "dedicatedNetworking": {
+                    "enabled": true,
+                    "includeStandby": true,
+                    "public": true,
+                    "serviceType": "LoadBalancer"
+                }
+            }
+        });
+        let params = PatchParams::apply("functional-test-dedicated-networking");
+        let patch = Patch::Apply(&coredb_json);
+        let _coredb_resource = coredbs.patch(name, &params, &patch).await.unwrap();
+
+        let service_dedicated = service_exists(
+            context.clone(),
+            &namespace,
+            &format!("{}-dedicated", name),
+            false,
+        )
+        .await;
+        let service_dedicated_ro = service_exists(
+            context.clone(),
+            &namespace,
+            &format!("{}-dedicated-ro", name),
+            false,
+        )
+        .await;
+
+        assert!(service_dedicated.is_some());
+        assert!(service_dedicated_ro.is_some());
+
+        let service = service_dedicated.unwrap();
+        assert_eq!(
+            service.spec.as_ref().unwrap().type_,
+            Some("LoadBalancer".to_string())
+        );
+        assert_eq!(
+            service
+                .metadata
+                .labels
+                .as_ref()
+                .expect("Labels should be present")
+                .get("public")
+                .expect("Public label should be present"),
+            "true"
+        );
+
+        let service = service_dedicated_ro.unwrap();
+        assert_eq!(
+            service.spec.as_ref().unwrap().type_,
+            Some("LoadBalancer".to_string())
+        );
+        assert_eq!(
+            service
+                .metadata
+                .labels
+                .as_ref()
+                .expect("Labels should be present")
+                .get("public")
+                .expect("Public label should be present"),
+            "true"
+        );
+
```

Comment:
please add an assertion of what the hostname should look like on the DNS annotation